Patches/OutroPatch.cs

```diff
@@ -25,35 +25,26 @@ public static void Postfix(AmongUsClient __instance, [HarmonyArgument(0)] ref En
 
         // if game is H&S or Host no have mod
         if (!GameStates.IsModHost || GameStates.IsHideNSeek) return;
-        
+
+        Logger.Info("-----------Game over-----------", "Phase");
+
         try
         {
             foreach (var pvc in GhostRoleAssign.GhostGetPreviousRole.Keys) // Sets role back to original so it shows up in /l results.
             {
-                var plr = Utils.GetPlayerById(pvc);
-                if (plr == null || !plr.GetCustomRole().IsGhostRole()) continue;
-
-                CustomRoles prevrole = GhostRoleAssign.GhostGetPreviousRole[pvc];
-                Main.PlayerStates[pvc].SetMainRole(prevrole);
-
-                if (AmongUsClient.Instance.AmHost)
-                {
-                    MessageWriter writer = AmongUsClient.Instance.StartRpcImmediately(PlayerControl.LocalPlayer.NetId, (byte)CustomRPC.SetCustomRole, SendOption.Reliable, -1);
-                    writer.Write(pvc);
-                    writer.WritePacked((int)prevrole);
-                    AmongUsClient.Instance.FinishRpcImmediately(writer);
-                }
+                if (!Main.PlayerStates.TryGetValue(pvc, out var state) || !state.MainRole.IsGhostRole()) continue;
+                if (!GhostRoleAssign.GhostGetPreviousRole.TryGetValue(pvc, out CustomRoles prevrole)) continue;
+                    
+                Main.PlayerStates[pvc].MainRole = prevrole;
```

Comment:
Seems it's not work for modded clients

Reply:
oh wait, that's true. I know the reason, gonna fix it.

Reply:
added a fix, to explain the reason I have changed this outropatch originally, is because when people got ghost-roles and get reverted. It would mark themselves having the default (-100) ability usage.